test/unit/user_test.rb

```diff
@@ -91,23 +92,23 @@ class UserTest < ActiveSupport::TestCase
     test 'user model cannot be created with 4 of the same repeated digits as pin' do
       @user.pin = "1111"
       @user.save
-      assert_equal(@user.errors.messages[:pin],["PIN does not meet our requirements. Please try again."])
+      assert_equal(@user.errors.messages[:pin],[@pin_msg])
     end
   end
 
   [generate_email].each do |new_email|
     test 'user model cannot be created with alternate repeated digits as pin' do
       @user.pin = "1212"
       @user.save
-      assert_equal(@user.errors.messages[:pin],["PIN does not meet our requirements. Please try again."])
+      assert_equal(@user.errors.messages[:pin],[@pin_msg])
     end
   end
 
   [generate_email].each do |new_email|
     test 'user model cannot be created with 3 of the same repeated digits in a row as pin' do
       @user.pin = "0007"
       @user.save
-      assert_equal(@user.errors.messages[:pin],["PIN does not meet our requirements. Please try again."])
+      assert_equal(@user.errors.messages[:pin],[@pin_msg])
     end
   end
 
```

Comment:
I'd like to see a couple more tests of validation — making sure a 3-character long PIN doesn't get through; a 32-character-long PIN does; a 33-character-long PING doesn't.